react/features/video-quality/subscriber.js

```diff
@@ -17,16 +17,47 @@ import { getMinHeightForQualityLvlMap } from './selector';
 declare var APP: Object;
 
 /**
- * StateListenerRegistry provides a reliable way of detecting changes to selected
- * endpoints state and dispatching additional actions. The listener is debounced
+ * Handles changes in the visible participants in the filmstrip. The listener is debounced
  * so that the client doesn't end up sending too many bridge messages when the user is
  * scrolling through the thumbnails prompting updates to the selected endpoints.
  */
 StateListenerRegistry.register(
-    /* selector */ state => state['features/video-layout'].selectedEndpoints,
-    /* listener */ debounce((selectedEndpoints, store) => {
+    /* selector */ state => state['features/filmstrip'].visibleParticipants,
+    /* listener */ debounce((visibleParticipants, store) => {
         _updateReceiverVideoConstraints(store);
-    }, 1000));
+    }, 100));
+
+/**
+ * Handles the use case when the on-stage participant has changed.
+ */
+StateListenerRegistry.register(
+    state => state['features/large-video'].participantId,
+    (participantId, store) => {
+        _updateReceiverVideoConstraints(store);
+    }
+);
+
+/**
+ * Handles the use case when we have set some of the constraints in redux but the conference object wasn't available
+ * and we haven't been able to pass the constraints to lib-jitsi-meet.
+ */
+StateListenerRegistry.register(
+    state => state['features/base/conference'].conference,
+    (conference, store) => {
+        _updateReceiverVideoConstraints(store);
```

Comment:
@hristoterezov Are you sure this will be called when the conference object get created ? Wouldn't this get triggered only when the conference object changes ? I was testing something else and I noticed that this wasn't working.

Reply:
I remember verifying it was working. will test it again.

Before we join the conference `state['features/base/conference'].conference` is `undefined` and on join we assign it to `state['features/base/conference'].conference`.

What do you mean by not working?

Reply:
@jallamsetty1 I put a console log there and I see it. So I suppose it is working!